webapp/sources/rudder/rudder-rest/src/test/scala/com/normation/rudder/rest/TestRestFromFileDef.scala

```diff
@@ -78,18 +61,47 @@ class TestRestFromFileDef extends TraitTestApiFromYamlFiles with AfterAll {
     orig.replaceAll("VALID-REV", revision.rev.value)
   }
 
-  val transformations: Map[String, String => String] = Map(
-    ("api_revisions.yml" -> copyTransformApiRevision _)
-  )
-
+  // you can pass a list of file to test exclusively if you don't want to test all .yml
+  // files in src/test/resource/${yamlSourceDirectory}
   // we are testing error cases, so we don't want to output error log for them
   org.slf4j.LoggerFactory
     .getLogger("com.normation.rudder.rest.RestUtils")
     .asInstanceOf[ch.qos.logback.classic.Logger]
     .setLevel(ch.qos.logback.classic.Level.OFF)
 
-  // you can pass a list of file to test exclusively if you don't want to test all .yml
-  // files in src/test/resource/${yamlSourceDirectory}
-  doTest(Nil)
+  val restTestSetUp = RestTestSetUp.newEnv
+  // update package management revision with new commits
+
+  // let's say that's /var/rudder/share
+  val tmpApiTemplate: File = restTestSetUp.baseTempDirectory / "apiTemplates"
+
+  val yamlSourceDirectory:  String = "api"
+  val yamlDestTmpDirectory: File   = tmpApiTemplate
+
+  val transformations: Map[String, String => String] = Map(
+    ("api_revisions.yml" -> copyTransformApiRevision(restTestSetUp) _)
+  )
+
+  tmpApiTemplate.createDirectories()
+
+  override def spec: Spec[TestEnvironment with Scope, Any] = {
+    (suite("All REST tests defined in files") {
+
+      for {
+        s <- TraitTestApiFromYamlFiles.doTest(
+               yamlSourceDirectory,
+               yamlDestTmpDirectory,
+               restTestSetUp,
+               Nil,
+               transformations
+             )
+        _ <- effectUioUnit(
+               if (java.lang.System.getProperty("tests.clean.tmp") != "false") IOResult.attempt(restTestSetUp.cleanup())
```

Comment:
there is also an "afterAll" annotation, but it feels rather natural to do like that